app/main.py

```diff
@@ -1,13 +1,42 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            }
+            for error in errors
+        ],
+        "path": file_path,
+        "status": "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "path": path,
+            "status": "passed" if not errors else "failed",
+            "errors": [{
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            } for error in errors]
+        } if errors else {"path": path, "status": "passed", "errors": []}
+        for path, errors in linter_report.items()
+    ]
```

Comment:
Do not duplicate code that is implemented in other functions. Call functions instead, e.g. you should call `format_single_linter_file` in the `format_linter_report` and `format_linter_error` in the `format_single_linter_file`.